libs/langchain/langchain/chat_models/huggingface_llama2.py

```diff
@@ -0,0 +1,188 @@
+import importlib.util
+from enum import Enum
+from typing import Any, Dict, List, Optional, Union
+
+from langchain.callbacks.manager import (
+    CallbackManagerForLLMRun,
+)
+from langchain.chat_models.base import BaseChatModel
+from langchain.pydantic_v1 import root_validator
+from langchain.schema import ChatResult
+from langchain.schema.messages import (
+    AIMessage,
+    BaseMessage,
+    HumanMessage,
+    SystemMessage,
+)
+from langchain.schema.output import ChatGeneration
+
+
+class InstructionTokens(Enum):
+    def __str__(self) -> str:
+        return self.value
+
+    B_INST = "[INST]"
+    E_INST = "[/INST]"
+
+
+class SystemTokens(Enum):
+    def __str__(self) -> str:
+        return self.value
+
+    B_SYS = "<<SYS>>"
+    E_SYS = "<</SYS>>"
+
+
+class ChatLlama2Hf(BaseChatModel):
```

Comment:
What about making this a generic `ChatHuggingFacePipeline` class that could be expanded as more chat models become available? Will defer to others for scope